nixos/modules/services/networking/iwd.nix

```diff
@@ -22,8 +148,29 @@ in {
 
     systemd.packages = [ pkgs.iwd ];
 
+    systemd.services.iwd.preStart = mkIf (cfg.networks != {}) (let
+      ssids = flip mapAttrsToList predefinedNetworks (file: content: {
+        inherit (content) file;
+        local = encodeIfNeeded file content.extension;
+      });
+    in ''
+      find /var/lib/iwd -type l -lname '${builtins.storeDir}/*' ${optionalString (ssids != {}) ''
+        -not \( ${concatMapStringsSep " -o " ({ local, ... }: "-name '${baseNameOf local}*'") ssids} \) \
+      ''} \
+        | xargs -I {} rm -f {}
```

Comment:
This seems really complicated and I'm having a hard time grasping what it actually does. I think some comments would help a lot with that. You could also use `find`'s `-exec` option to simplify this a bit.

Reply:
If you just want to delete stuff you can also use find's builtin `-delete`.

Reply:
To summarize, this removes all files in `/var/lib/iwd` that are a symlink to a store-path, but **aren't** defined in `cfg.networks`.